js.go

```diff
@@ -2558,7 +2573,8 @@ func (sub *Subscription) Fetch(batch int, opts ...PullOpt) ([]*Msg, error) {
 		// are no messages.
 		msg, err = sub.nextMsgWithContext(ctx, true, false)
 		if err != nil {
-			if err == errNoMessages {
+			// We skip both 404 and any type of 408 errors at this stage.
+			if err == errNoMessages || err == errRequestsPending {
```

Comment:
This will have no effect. We just called sub.nextMsgWithContext(), which is a low-level NATS call, but since we indicate that this is an internal pull call, it will return errNoMessages if there was no message and not go into a wait. You can check that function for details, but there is no way this function returns errRequestsPending since I don't see that you added that outside of checkMsg().

Reply:
thanks for the review, removed the check since it had no effect